File: Roman_to_Integer/roman_to_integer.cpp
#include <iostream>
#include <string>
#include <map>
using namespace std;

class Solution {
public:
    int romanToInt(string s) {
        map<char,int> m{{'M',1000}, {'D',500}, {'C',100}, {'L',50}, {'X', 10},{'V',5},{'I',1}};
        int count = 0;
        int prev = 1e4;
        for(char i:s)
        {
            count += m[i];
            if(prev < m[i])
            {
                count -= 2*prev;
            }
            prev = m[i];
        }
        return count;     
    }
};

int main() {
    Solution solution;
    string roman = "MCMXCIV"; // Example Roman numeral
    int integer = solution.romanToInt(roman);
    cout << "The integer value of " << roman << " is: " << integer << endl;
    return 0;
}

File: Roman_to_Integer/roman_to_integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        m = {'M':1000, 'D':500, 'C':100, 'L':50, 'X': 10,'V':5,'I':1}
        count = 0
        prev = 0
        for c in s:
            count += m[c]
            if prev < m[c]:            
                count = count - 2*prev
            prev = m[c]
        return count; 


# Example usage:
if __name__ == "__main__":
    s = Solution()
    print(s.romanToInt("III"))  # Output: 3
    print(s.romanToInt("IV"))   # Output: 4
    print(s.romanToInt("IX"))   # Output: 9
    print(s.romanToInt("LVIII")) # Output: 58
    print(s.romanToInt("MCMXCIV")) # Output: 1994
    print(s.romanToInt("MMXXIII")) # Output: 2023